ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py

```diff
@@ -0,0 +1,332 @@
+				fieldtype="Data",
+				insert_after="zenoti_item_category",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="zenoti_item_type",
+				label="Zenoti Item Type",
+				fieldtype="Select",
+				options="\nRetail\nConsumable\nBoth",
+				insert_after="zenoti_bussiness_unit_id",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+		"Sales Invoice": [
+			dict(
+				fieldname="zenoti_invoice_no",
+				label="Zenoti Invoice No",
+				fieldtype="Data",
+				insert_after="naming_series",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="zenoti_receipt_no",
+				label="Zenoti Receipt No",
+				fieldtype="Data",
+				insert_after="zenoti_invoice_no",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+		"Purchase Order": [
+			dict(
+				fieldname="zenoti_order_no",
+				label="Zenoti Order No",
+				fieldtype="Data",
+				insert_after="naming_series",
+				read_only=1,
+				print_hide=1,
+			)
+		],
+		"Purchase Invoice": [
+			dict(
+				fieldname="zenoti_order_no",
+				label="Zenoti Order No",
+				fieldtype="Data",
+				insert_after="naming_series",
+				read_only=1,
+				print_hide=1,
+			)
+		],
+		"Employee": [
+			dict(
+				fieldname="zenoti_employee_id",
+				label="Zenoti Employee Id",
+				fieldtype="Data",
+				insert_after="naming_series",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="zenoti_employee_code",
+				label="Zenoti Employee Code",
+				fieldtype="Data",
+				insert_after="zenoti_employee_id",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="zenoti_employee_username",
+				label="Zenoti Employee Username",
+				fieldtype="Data",
+				insert_after="zenoti_employee_code",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+		"Sales Invoice Item": [
+			dict(
+				fieldname="zenoti_employee_details",
+				label="Zenoti Employee Details",
+				fieldtype="Section Break",
+				insert_after="delivered_by_supplier",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="sold_by",
+				label="Sold By",
+				fieldtype="Link",
+				options="Employee",
+				insert_after="zenoti_employee_details",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="tips_column_break",
+				label="",
+				fieldtype="Column Break",
+				options="Employee",
+				insert_after="sold_by",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="tips",
+				label="Tips",
+				fieldtype="Data",
+				insert_after="tips_column_break",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+		"Stock Entry": [
+			dict(
+				fieldname="zenoti_order_id",
+				label="Zenoti Order Id",
+				fieldtype="Data",
+				insert_after="naming_series",
+				read_only=1,
+				print_hide=1,
+				hidden=1,
+			),
+			dict(
+				fieldname="zenoti_order_no",
+				label="Zenoti Order No",
+				fieldtype="Data",
+				insert_after="zenoti_order_id",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+	}
+
+	create_custom_fields(custom_fields, update=False)
```

Comment:
`update=False` Intentional?

Reply:
Yes I don't want the fields to be updated every time this document is saved.

Reply:
Hmm. I should also do this and make that flag configurable so we can pass `True` from patches when custom fields do change. 👍